homeassistant/components/music_assistant/media_player.py

```diff
@@ -434,6 +493,41 @@ async def _async_handle_play_announcement(
             self.player_id, url, use_pre_announce, announce_volume
         )
 
+    @catch_musicassistant_error
+    async def _async_handle_transfer_queue(
+        self, source_player: str | None = None, auto_play: bool | None = None
+    ) -> None:
+        """Transfer the current queue to another player."""
+        if not source_player:
+            # no source player given; try to find a playing player(queue)
+            for queue in self.mass.player_queues:
+                if queue.state == MassPlayerState.PLAYING:
+                    source_queue_id = queue.queue_id
+                    break
+            else:
+                raise HomeAssistantError(
+                    "Source player not specified and no playing player found."
+                )
+        else:
+            # resolve HA entity_id to MA player_id
+            if (hass_state := self.hass.states.get(source_player)) is None:
```

Comment:
I'm honestly not even sure why we are quering the states object here?

We can just do `entity := entity_registry.async_get(source_player)` in the code below directly?

Reply:
yeah, silly mistake.